src/ol_concourse/pipelines/open_edx/open_edx_plugins/build_publish_plugins.py

```diff
@@ -0,0 +1,139 @@
+    TaskStep,
+)
+from ol_concourse.lib.resource_types import pypi_resource
+from ol_concourse.lib.resources import git_repo, pypi
+
+plugin_dict = {
+    "src/edx_sysadmin": {
+        "target_name": "edx_sysadmin_package",
+        "pypi_package_name": "edx-sysadmin",
+    },
+    "src/ol_openedx_canvas_integration": {
+        "target_name": "canvas_integration_package",
+        "pypi_package_name": "ol-openedx-canvas-integration",
+    },
+    "src/ol_openedx_checkout_external": {
+        "target_name": "checkout_external_package",
+        "pypi_package_name": "ol-openedx-checkout-external",
+    },
+    "src/ol_openedx_course_export": {
+        "target_name": "course_export_package",
+        "pypi_package_name": "ol-openedx-course-export",
+    },
+    "src/ol_openedx_course_structure_api": {
+        "target_name": "course_structure_api_package",
+        "pypi_package_name": "ol-openedx-course-structure-api",
+    },
+    "src/ol_openedx_git_auto_export": {
+        "target_name": "edx_git_auto_export",
+        "pypi_package_name": "ol-openedx-git-auto-export",
+    },
+    "src/ol_openedx_logging": {
+        "target_name": "logging_package",
+        "pypi_package_name": "ol-openedx-logging",
+    },
+    "src/ol_openedx_otel_monitoring": {
+        "target_name": "otel_monitoring_package",
+        "pypi_package_name": "ol-openedx-otel-monitoring",
+    },
+    "src/ol_openedx_rapid_response_reports": {
+        "target_name": "rapid_response_plugin_dist",
+        "pypi_package_name": "ol-openedx-rapid-response-reports",
+    },
+    "src/ol_openedx_sentry": {
+        "target_name": "sentry_package",
+        "pypi_package_name": "ol-openedx-sentry",
+    },
+}
+
+git_resource_list = []
+pypi_resource_list = []
+job_list = []
+for path, config in plugin_dict.items():
+    git_resource_list.append(
+        git_repo(
+            Identifier(f"{config['target_name']}-repo"),
+            uri="https://github.com/mitodl/open-edx-plugins",
+            paths=[path],
+            check_every="24h",
+        )
+    )
+    pypi_resource_list.append(
+        pypi(
+            Identifier(f"{config['target_name']}-pypi-package"),
+            package_name=config["pypi_package_name"],
+            username="((pypi_creds.username))",
+            password="((pypi_creds.password))",  # noqa: S106
+        )
+    )
+    job_list.append(
+        Job(
+            name=f"build-{config['target_name']}",
+            plan=[
+                GetStep(
+                    get=git_resource_list[-1].name,
+                    trigger=True,
+                ),
+                TaskStep(
+                    task=Identifier(f"pants-package-{config['target_name']}"),
+                    config=TaskConfig(
+                        platform=Platform.linux,
+                        image_resource=AnonymousResource(
+                            type=REGISTRY_IMAGE,
+                            source=RegistryImage(repository="python", tag="3.9"),
```

Comment:
Should this maybe come from a `LoadVar` step that extracts the version from the git repo? Like the pyproject.toml file?

Reply:
I don't follow what you're getting at here but in the interest of getting it done I'm going to leave it be for now 👍